src/Xamarin.Android.Tools.AndroidSdk/AndroidSdkInfo.cs

```diff
@@ -185,5 +185,24 @@ public static void DetectAndSetPreferredJavaSdkPathToLatest (Action<TraceLevel,
 			var sdk = CreateSdk (logger);
 			sdk.SetPreferredJavaSdkPath (latestJdk.HomePath);
 		}
+
+		public static string GetPreferredAndroidToolsPath(string androidSdkPath)
+		{
+			var toolsDir = Path.Combine (androidSdkPath, "cmdline-tools");
+			if (!Directory.Exists (toolsDir)) {
+				// Trying to fallback to "tools"
+				var oldToolsDir = Path.Combine (androidSdkPath, "tools");
+				if (Directory.Exists (oldToolsDir))
+					return oldToolsDir;
+
+				return Path.Combine(toolsDir, "latest");
+			}
+
+			if (Directory.Exists (Path.Combine (toolsDir, "latest")))
+				return Path.Combine (toolsDir, "latest");
+
+			var latestToolsDir = Directory.GetDirectories (toolsDir).Max ();
```

Comment:
This isn't quite semantically correct, because `Max()` sorts *lexicographically*, not ordinally:

```csharp
Console.WriteLine (new[]{"1.0", "2.0", "11.0"}.Max ());
// Prints "2.0", because '2' > '1'
```

Fortunately `latest` *always* sorts as the "highest" value…

Reply:
@jonathanpeppers that is correct, will fix it.